contracts/deploy/base/009_upgrade_vault.js

```diff
@@ -0,0 +1,77 @@
+const { deployOnBaseWithGuardian } = require("../../utils/deploy-l2");
+const { deployWithConfirmation } = require("../../utils/deploy");
+const addresses = require("../../utils/addresses");
+const { utils } = require("ethers");
+
+module.exports = deployOnBaseWithGuardian(
+  {
+    deployName: "009_upgrade_vault",
+  },
+  async ({ ethers }) => {
+    const cOETHbVaultProxy = await ethers.getContract("OETHBaseVaultProxy");
+    const cOETHbVault = await ethers.getContractAt(
+      "IVault",
+      cOETHbVaultProxy.address
+    );
+    const cOETHbDripperProxy = await ethers.getContract("OETHBaseDripperProxy");
+
+    // Deploy new implementation
+    const dOETHbVaultCore = await deployWithConfirmation("OETHBaseVaultCore", [
+      addresses.base.WETH,
+    ]);
+    const dOETHbVaultAdmin = await deployWithConfirmation("OETHBaseVaultAdmin");
+
+    return {
+      actions: [
+        {
+          // 1. Upgrade VaultCore
+          contract: cOETHbVaultProxy,
+          signature: "upgradeTo(address)",
+          args: [dOETHbVaultCore.address],
+        },
+        {
+          // 2. Upgrade VaultAdmin
+          contract: cOETHbVault,
+          signature: "setAdminImpl(address)",
```

Comment:
we don't need to call this one right?

Reply:
We actually have to. Async withdrawal code got merged after we deployed OETHb. And that VaultAdmin adds a "setDripper" function. We need to set it so that VaultCore won't break on rebase (it's set as last step on this file).

Reply:
Aha I see changes out of scope of this PR 👍